actions/schema-up/index.js

```diff
@@ -0,0 +1,111 @@
+
+  const {exitCode: hasSchemaChanges} = await tools.runInWorkspace(
+    'git', ['diff', '--quiet', '--', 'graphql/schema.graphql'],
+    {reject: false},
+  );
+  if (hasSchemaChanges === 0) {
+    tools.log.info('No schema changes to fetch.');
+    tools.exit.neutral('Nothing to do.');
+  }
+
+  tools.log.info('Committing schema changes.');
+  await tools.runInWorkspace('git', ['commit', '--all', '--message=":arrow_up: GraphQL schema"']);
+
+  tools.log.info('Re-running relay compiler.');
+  const {failed: relayFailed, all: relayOutput} = await tools.runInWorkspace(
+    path.resolve(__dirname, 'node_modules', '.bin', 'relay-compiler'),
+    ['--src', './lib', '--schema', 'graphql/schema.graphql'],
+    {reject: false},
+  );
+
+  const {exitCode: hasRelayChanges} = await tools.runInWorkspace(
+    'git', ['diff', '--quiet', '--', '**/__generated__/*.graphql.js'],
+    {reject: false},
+  );
+
+  if (hasRelayChanges === 0 && !relayFailed) {
+    tools.log.info('Generated relay files are unchanged.');
+    await tools.runInWorkspace('git', ['push']);
+    tools.exit.success('Schema is up to date on master.');
+  }
+
+  tools.log.info('Checking for unmerged schema update pull requests.');
+  const {data: openSchemaUpdatePRs} = await tools.github.graphql(`
+    query OpenPullRequests($owner: String!, $repo: String!) {
+      repository(owner: $owner, name: repo) {
+        pullRequests(first: 1, states: [OPEN], labels: ["schema update"]) {
+          totalCount
+        }
+      }
+    }
+  `, {...tools.context.repo});
+
+  if (openSchemaUpdatePRs.repository.pullRequests.totalCount > 0) {
+    tools.exit.failure('One or more schema update pull requests are already open. Please resolve those first.');
+  }
+
+  const branchName = `schema-update/${Date.now()}`;
+  tools.log.info(`Commiting relay-compiler changes to a new branch ${branchName}.`);
+  await tools.runInWorkspace('git', ['checkout', '-b', branchName]);
+  if (!relayFailed) {
+    await tools.runInWorkspace('git', ['commit', '--all', '--message=":gear: relay-compiler changes"']);
+  }
+  await tools.runInWorkspace('git', ['push', 'origin', branchName]);
+
+  tools.log.info('Creating a pull request.');
+
+  let body = `:robot: _This automated pull request brought to you by [a GitHub action](/actions/schema-up)_ :robot:
+
+The GraphQL schema has been automatically updated and \`relay-compiler\` has been re-run on the package source.`;
+
+  if (!relayFailed) {
+    body += ' The modified files have been committed to this branch and pushed. ';
+    body += 'If all of the tests pass in CI, merge with confidence :zap:';
+  } else {
+    body += ' `relay-compiler` failed with the following output:\n\n```\n';
+    body += relayOutput;
+    body += '\n```\n\nCheck out this branch to fix things so we don\'t break.';
+  }
+
+  const {data: createPR} = await tools.github.graphql(`
+    mutation CreatePullRequest($owner: String!, $repo: String!, $body: String!, $headRefName: String!) {
```

Comment:
Looks like the required input for `createPullRequest` is:

<img width="562" alt="GraphQL_API_Explorer___GitHub_Developer_Guide" src="https://user-images.githubusercontent.com/7910250/57044103-42f9da00-6c1e-11e9-96d6-428d21e0c749.png">

So we also need `repositoryId` and `baseRefName`.

```suggestion
    mutation CreatePullRequest($repositoryId: ID!, $baseRefName: String!, $body: String!, $headRefName: String!) {
```